### PR TITLE
EVA-2605  Group RS count by taxonomy/assembly and update clustering release tracker table

### PR DESCRIPTION
**Pull request by nitin-ebi**

### CODE REVIEW

File: eva-accession-clustering-automation/clustering_automation/create_clustering_progress_table.py [hunk 1]
```diff
@@ -100,19 +102,62 @@ def fill_in_from_previous_inventory(private_config_xml_file, release_version):
             execute_query(pg_conn, query_insert)
 
 
+def fill_num_rs_id_for_taxonomy_and_assembly(mongo_source, private_config_xml_file):
```
Comment (tcezard) [inline eva-accession-clustering-automation/clustering_automation/create_clustering_progress_table.py:105]: Would it be possible to limit the run of the script to a taxonomy id and retrieve the assembly accession from the EVAPRO ?

File: eva-accession-clustering-automation/clustering_automation/create_clustering_progress_table.py [hunk 2]
```diff
@@ -100,19 +102,62 @@ def fill_in_from_previous_inventory(private_config_xml_file, release_version):
...
+    aggregated_rs_info = cve_collection.aggregate([group_rs_by_taxonomy_and_assembly], batchSize=0)
+    for agg_info in aggregated_rs_info:
+        with get_metadata_connection_handle("development", private_config_xml_file) as pg_conn:
+            taxonomy = agg_info["_id"]["taxonomy"]
+            assembly = agg_info["_id"]["assembly"]
+            rs_count = agg_info["rs_count"]
+            logger.info(f'updating rs count({rs_count}) for taxonomy({taxonomy}) and assembly({assembly})')
+            query_update = (
+                f'UPDATE eva_progress_tracker.clustering_release_tracker SET num_rs_to_release={rs_count} '
+                f'WHERE taxonomy={taxonomy} and assembly_accession=\'{assembly}\'')
```
Comment (tcezard) [inline eva-accession-clustering-automation/clustering_automation/create_clustering_progress_table.py:123]: In some case, the row won't exists already so this update won't work. You'll have to create the whole row in that case.

File: eva-accession-clustering-automation/clustering_automation/create_clustering_progress_table.py (hunk 1 above)
Comment (nitin-ebi) [reply eva-accession-clustering-automation/clustering_automation/create_clustering_progress_table.py]: done

File: eva-accession-clustering-automation/clustering_automation/create_clustering_progress_table.py (hunk 2 above)
Comment (nitin-ebi) [reply eva-accession-clustering-automation/clustering_automation/create_clustering_progress_table.py]: done

File: eva-accession-clustering-automation/clustering_automation/create_clustering_progress_table.py [hunk 3]
```diff
@@ -100,20 +103,152 @@ def fill_in_from_previous_inventory(private_config_xml_file, release_version):
...
+                        WHERE taxonomy={taxonomy} and assembly_accession='{assembly}'""")
+    execute_query(pg_conn, query_update)
+
+
+def insert_new_entry_for_taxonomy_assembly(pg_conn, sources, rs_count, release_version, taxonomy, assembly, reference_directory):
+    logger.info(f'inserting rs count({rs_count}) for taxonomy({taxonomy}) and assembly({assembly})')
+    scientific_name = get_scientific_name(pg_conn, taxonomy)
+    release_folder_name = normalise_taxon_scientific_name(scientific_name)
+    fasta_path = NCBIAssembly(assembly, scientific_name, reference_directory).assembly_fasta_path
+    report_path = NCBIAssembly(assembly, scientific_name, reference_directory).assembly_report_path
```
Comment (tcezard) [inline eva-accession-clustering-automation/clustering_automation/create_clustering_progress_table.py:197]: My fault I should have been more explicit:
```suggestion
    ncbi_assembly = NCBIAssembly(assembly, scientific_name, reference_directory)
    fasta_path = ncbi_assembly.assembly_fasta_path
    report_path = ncbi_assembly.assembly_report_path
```

File: eva-accession-clustering-automation/clustering_automation/create_clustering_progress_table.py [hunk 4]
```diff
@@ -100,20 +103,152 @@ def fill_in_from_previous_inventory(private_config_xml_file, release_version):
...
+    if assembly in cve_res and assembly in dbsnp_res:
+        return ['DBSNP, EVA', cve_res[assembly] + dbsnp_res[assembly]]
+    elif assembly in cve_res and assembly not in dbsnp_res:
+        return ['EVA', cve_res[assembly]]
+    elif assembly not in cve_res and assembly in dbsnp_res:
+        return ['DBSNP', dbsnp_res[assembly]]
+    else:
+        # if there is an assembly in evapro but there is no clustered variant in any of the collections,
+        # an entry will still be created in clustering tracking table with sources as None and rs count 0
+        return ['None', 0]
```
Comment (tcezard) [inline eva-accession-clustering-automation/clustering_automation/create_clustering_progress_table.py:174]: For this case I would avoid creating a row at all.

Comment (nitin-ebi) [reply eva-accession-clustering-automation/clustering_automation/create_clustering_progress_table.py]: done

File: eva-accession-clustering-automation/clustering_automation/create_clustering_progress_table.py (hunk 3 above)
Comment (nitin-ebi) [reply eva-accession-clustering-automation/clustering_automation/create_clustering_progress_table.py]: updated

**Review by tcezard — APPROVED** [review]

**Pull request merged**